meson.build

```diff
@@ -969,7 +969,7 @@ if tests_enable_stack_protector
   endif
 endif
 
-test_env = environment()
+test_env = environment({'PICOLIBC_TEST' : '1'})
```

Comment:
After a few years,  this will look like a mystery. I think you should mention at this point why the script is setting the variable. You could just copy the commit message here:
```

Meson version 0.53.2 (at least) doesn't check for the
skip_sanity_check value before attempting to run test fragments during
configuration. That means the exe_wrapper value needs to handle being
run at that point. To support this, the exe_wrapper becomes a script
which checks for a environment variable (PICOLIBC_TEST) and exits
indicating success if that is not present.
```
You could also mention that this has been fixed in Meson 0.54.2, in order to facilitate removing this work-around  after a while.

Reply:
that's a really good idea. thanks!